### PR TITLE
Add some config management commands

### LINKED ISSUE #111: Make initial config file creation easier
Ok so the config file is already pretty easy to make:  https://github.com/pulp/pulp-cli#configuration It would be easier still though if the user didn't have to make the dir and then then the config file. If they could just have to edit it to customize it instead of the default that would be easier.

Does it make any sense to do any of the following?

a) have the config file created by default on installation.

b) have a CLI command that would populate the config file (kind of like pulp smash has).

What do you think?

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #111

### CODE REVIEW

File: pulpcore/cli/core/config.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        verify_ssl = (
+            click.prompt(
+                "Verify SSL Connection",
+                type=click.Choice(["yes", "no"], case_sensitive=False),
+                default="yes",
+            )
+            == "yes"
+        )
+
+    if not overwrite and Path(location).exists():
```
Comment (mdellweg) [inline pulpcore/cli/core/config.py:58]: we should check before asking questions.

File: pulpcore/cli/core/config.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    username: Optional[str],
+    password: Optional[str],
+    verify_ssl: bool,
+    format: str,
+) -> None:
+    if interactive:
+        location = click.prompt("Config location", default=LOCATION)
+        base_url = click.prompt("API base url", default="https://localhost")
+        username = click.prompt("Username", default="")
+        password = click.prompt("Password", default="")
```
Comment (mdellweg) [inline pulpcore/cli/core/config.py:43]: We could fetch the defaults from the context below, i think. It would guarantee some consistency.

**Comment by mdellweg** [comment]
Would you mind adding a `pulp config edit` command that just opens the existing config file in the editor, and maybe preseeding it in case it was nonexisting?

File: pulpcore/cli/core/config.py (hunk 2 above)
Comment (daviddavis) [reply pulpcore/cli/core/config.py:43]: Which context?

**Comment by daviddavis** [comment]
> Would you mind adding a pulp config edit command that just opens the existing config file in the editor, and maybe preseeding it in case it was nonexisting?

Sure, yes. I think though it should raise an error if the file doesn't exist. I think the job of creating and seeding the file belongs to `create`.

File: docs/install.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+```
+
+### Validate
+
+To validate an existing config:
+
+```
+pulp config validate
+pulp config validate --strict  # validates that all settings are present
+```
```
Comment (mdellweg) [inline docs/install.md:56]: We could add a note that connectivity can be checked with `pulp status`, and may be that validation here is _only_ static analysis (is there a user friendly term for this?).

File: docs/install.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+machine localhost
+login admin
+password password
+```
+
+### Katello
+
+If you have a Katello environment and wish to use pulp-cli to connect to Pulp, you'll need to
+configure client certificate authentication:
```
Comment (mdellweg) [inline docs/install.md:73]: Not for this PR, but are we sure, that providing the cert will not collide with an existing netrc config?

File: pulpcore/cli/core/config.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        location = click.prompt("Config file location", default=LOCATION)
+        _check_location(location)
+        for setting in SETTINGS:
+            settings[setting] = prompt_config_option(setting)
+    else:
+        _check_location(location)
+        for setting in SETTINGS:
+            settings[setting] = locals()[setting] or ""
+
+    output = f"[cli]\n{toml.dumps(settings)}"
```
Comment (mdellweg) [inline pulpcore/cli/core/config.py:64]: This looks hacky to me, not being a toml expert.
Would this also work?
```suggestion
    output = toml.dumps({"cli": settings})
```

File: pulpcore/cli/core/config.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            raise click.ClickException("No output from editor. Aborting.")
+
+    Path(location).parent.mkdir(parents=True, exist_ok=True)
+    with Path(location).open("w") as sfile:
+        sfile.write(output)
+
+    click.echo(f"Created config file at '{location}'.")
+
+
+@config.command(help="Open the settings config file in an editor")
```
Comment (mdellweg) [inline pulpcore/cli/core/config.py:78]: Oh, you have peeked into my gettext experiments? :partying_face:

File: tests/scripts/test_config.sh [hunk 7]
```diff
@@ -22,3 +31,16 @@ expect_fail pulp --username test --password test --client "/some/path" status
...
+
+# CONFIG COMMAND
+
+expect_fail pulp config edit --location $test_settings
+
+expect_succ pulp config create --location $test_settings --base-url "http://testing"
+cat $test_settings | grep 'base_url = "http://testing"'
+cat $test_settings | grep 'verify_ssl = true'
+
+expect_fail pulp config create --location $test_settings
```
Comment (mdellweg) [inline tests/scripts/test_config.sh:44]: `make me a sandwich`
`access denied`
`sudo make me a sandwich`

```suggestion
expect_fail pulp config create --location $test_settings
expect_succ pulp config create --location $test_settings --overwrite
```

File: pulpcore/cli/core/config.py (hunk 5 above)
Comment (daviddavis) [reply pulpcore/cli/core/config.py:64]: I must have looked at a bad example as this does work.

File: docs/install.md (hunk 4 above)
Comment (daviddavis) [reply docs/install.md:73]: Yes, I will investigate.

File: pulpcore/cli/core/config.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/cli/core/config.py]: I was thinking about this `ctx.default_map`.

https://github.com/pulp/pulp-cli/blob/develop/pulpcore/cli/common/__init__.py#L23

File: docs/install.md (hunk 4 above)
Comment (daviddavis) [reply docs/install.md:80]: I filed an issue so I wouldn't forget https://github.com/pulp/pulp-cli/issues/135

**Review by mdellweg — APPROVED** [review]

**Pull request merged**